src/content/resources/glossary.md

```diff
@@ -336,3 +336,51 @@ class A implements B {} // A is a subtype of B, but NOT a subclass of B.
 
 class C extends D {} // C is a subtype AND a subclass of D.
 ```
+
+## Variance and variance positions
+
+A type parameter of a class (or a mixin, etc) is said to be _covariant_
+when the type as a whole "co-varies" with the actual type argument, that
+is: if we replace the type argument by a subtype then the type as a whole
+is also a subtype.
+
+For example, the type parameter of the class `List` is covariant because
+list types co-vary with their type argument: `List<int>` is a subtype of
+`List<Object>` because `int` is a subtype of `Object`.
+
+In Dart, all type parameters of all class, mixin, mixin class, and enum
+declarations are covariant.
+
+However, function types are different: A function type is covariant in the
+return type, but the opposite (known as _contravariant_) in its parameter
+types.  For example, the type `int Function(int)` is a subtype of the type
+`Object Function(int)`, but it is a supertype of `int Function(Object)`.
+
+This makes sense if you consider their substitutability (see the section
+about subtypes): If you expect at compile time to work with a function of
+type `int Function(int)` then it is OK to actually work with a function of
+type `int Function(Object)` at run time: You will pass an `int` to it (and
+that's fine, it accepts any `Object`), and it returns an `int` (which is
+exactly what you'd expect). Hence, `int Function(Object)` is a subtype of
+`int Function(int)`. Note that everything is turned upside-down when we
+consider a parameter type.
+
+When we consider a more complex type like `List<int Function(int)>` we need
+to talk about the _positions_ in the type. This is basically just a matter
+of turning one of the parts of the type into a placeholder (let's use `_`
+for that), and then consider what happens to the type when we put different
+types into that location in the type.
+
+For example, consider `List<int Function(_)>` as a template for a type
+where we can put different types into the location where `_` occurs.
+
+This type is contravariant in that position: For example, 
```

Comment:
Consider combining these two paragraphs, as the second one heavily relies on the first one. With the separation, it might not be clear what "This type" or "that position" refers to.

Reply:
Also is the following example meant to explain _why_ its contravariant in that position or is it _because_ it's contravariant in that position?

My understanding is that it's an example of understanding _why_ that position is contravariant, but it's not clear to me from reading. Perhaps flipping the order of the because clauses and starting at the `Object` and `int` parameter type relation, might make that more clear.

Reply:
Combined the two paragraphs.

I changed the return type from `int` to `void` in the following examples, in order to be able to talk about `int` as a type that is unambiguously associated with a single position in the type (avoiding the need to say "the first `int`" vs. "the second `int`" and such).

> is the following example meant to explain _why_ its contravariant in that position

The text is intended to explain how 'contravariant position' is defined, by example. This is in itself an example, because it is intended to define 'contravariant position' as well as 'covariant position', based on a generalization which the reader would probably perform automatically, based on the previous explanations. I added a paragraph where 'covariant position' is defined briefly, but explicitly, in order to improve on this situation.

The text defines 'contravariant position' by explaining why we must conclude that the overall type `List<void Function(_)>` varies in the opposite direction of the variation at `_`, and then states that this by definition makes the position of `_` contravariant.